decimal.go

```diff
@@ -0,0 +1,42 @@
+package luno
+
+import (
+	"strconv"
+	"strings"
+)
+
+type Decimal struct {
+	f         float64
+	precision int
+}
+
+func NewDecimal(f float64, precision int) Decimal {
+	return Decimal{f: f, precision: precision}
+}
+
+func (d Decimal) MarshalJSON() ([]byte, error) {
+	return []byte(`"` + d.String() + `"`), nil
+}
+
+func (d *Decimal) UnmarshalJSON(b []byte) error {
+	s := strings.Trim(string(b), `"`)
+	f, err := strconv.ParseFloat(s, 64)
+	if err != nil {
+		return err
+	}
+	*d = NewDecimal(f, getPrecision(f))
```

Comment:
I think the precision should come from the original string, not the float, because the float conversion is inexact. E.g. "0.1" -> 0.1000000000001 -> precision=12 instead of 1.

Reply:
OK - I don't think we'd be able to support float strings like "1.2e8" then, which has precision 0 but has 3 characters after the decimal point.

Reply:
True, should be fine though since the API never outputs in that format. (We should document the decimal type in the docs.)